src/views/projectExplorer/library.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * (c) Copyright IBM Corp. 2023
+ */
+
+import { ThemeIcon, TreeItemCollapsibleState, WorkspaceFolder } from "vscode";
+import { ProjectExplorerTreeItem } from "./projectExplorerTreeItem";
+import { getInstance } from "../../ibmi";
+import ObjectFile from "./objectFile";
+import MemberFile from "./memberFile";
+import { ContextValue } from "../../projectExplorerApi";
+
+export default class Library extends ProjectExplorerTreeItem {
+  static contextValue = ContextValue.library;
+  name: string;
+
+  constructor(public workspaceFolder: WorkspaceFolder, path: string, name: string) {
+    super(name, TreeItemCollapsibleState.Collapsed);
+    this.description = path;
+    this.contextValue = this.contextValue;
+    this.iconPath = new ThemeIcon(`symbol-folder`);
+    this.name = name;
+    this.tooltip = `Library ${path}`
+  }
+
+  async getChildren(): Promise<ProjectExplorerTreeItem[]> {
+    let items: ProjectExplorerTreeItem[] = [];
+
+    const ibmi = getInstance();
+    const files = await ibmi?.getContent().getObjectList({
+      library: this.name
+    });
+    if (files) {
+      for (const file of files) {
+        const path = `/QSYS.LIB/${this.name}/${file.name}`;
```

Comment:
consider that we will have to support iASP
/<iasp/QSYS.LIB/....
@worksofliam is there some code that Sanjula should be modelling off of

Reply:
You probably will have a shared function to build this path and put it in the `ibmi` object?

Reply:
Planning to add iASP support in a separate PR. Issue to track this: #6